plugin/mysql_extend/init.go

```diff
@@ -0,0 +1,185 @@
+	}
+
+	m.mysqlConfig = mysqlConfig
+	for c.Next() {
+		for c.NextBlock() {
+			switch c.Val() {
+			case "dsn":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.dsn = c.Val()
+			case "dump_file":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.dumpFile = c.Val()
+			case "ttl":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userTTL, err := strconv.Atoi(c.Val())
+				if err != nil || userTTL <= zero {
+					m.ttl = defaultTTL
+				} else {
+					m.ttl = uint32(userTTL)
+				}
+			case "zones_table":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.zonesTable = c.Val()
+			case "records_table":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.recordsTable = c.Val()
+			case "db_max_idle_conns":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userMaxIdleConns, err := strconv.Atoi(c.Val())
+				if err != nil || userMaxIdleConns <= zero {
+					m.maxIdleConns = defaultMaxIdleConns
+				} else {
+					m.maxIdleConns = userMaxIdleConns
+				}
+			case "db_max_open_conns":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userMaxOpenConns, err := strconv.Atoi(c.Val())
+				if err != nil || userMaxOpenConns <= zero {
+					m.maxOpenConns = defaultMaxOpenConns
+				} else {
+					m.maxOpenConns = userMaxOpenConns
+				}
+			case "db_conn_max_idle_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userConnMaxIdleTime, err := time.ParseDuration(c.Val())
+				if err != nil || userConnMaxIdleTime <= zeroTime {
+					m.connMaxIdleTime = defaultConnMaxIdleTime
+				} else {
+					m.connMaxIdleTime = userConnMaxIdleTime
+				}
+			case "db_conn_max_life_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userConnMaxLifeTime, err := time.ParseDuration(c.Val())
+				if err != nil || userConnMaxLifeTime <= zeroTime {
+					m.connMaxLifetime = defaultConnMaxLifeTime
+				} else {
+					m.connMaxLifetime = userConnMaxLifeTime
+
+				}
+			case "fail_heartbeat_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userFailHeartBeatTime, err := time.ParseDuration(c.Val())
+				if err != nil || userFailHeartBeatTime <= zeroTime {
+					m.failHeartbeatTime = defaultFailHeartBeatTime
+				} else {
+					m.failHeartbeatTime = userFailHeartBeatTime
+				}
+			case "success_heartbeat_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userSuccessHeartBeatTime, err := time.ParseDuration(c.Val())
+				if err != nil || userSuccessHeartBeatTime <= zeroTime {
+					m.successHeartbeatTime = defaultSuccessHeartBeatTime
+				} else {
+					m.successHeartbeatTime = userSuccessHeartBeatTime
+				}
+
+			case "query_zone_sql":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.queryZoneSQL = c.Val()
+			case "query_record_sql":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.queryRecordSQL = c.Val()
+			default:
+				return c.Errf("unknown property '%s'", c.Val())
+			}
+		}
+	}
+	return nil
+}
+
+func (m *Mysql) createTables() {
```

Comment:
I don't think the plugin should create the tables. The DNS server should not have any write access to the database.

Reply:
This is just for the convenience of development. Although it has no special impact on the DB, it is not very good overall and does not conform to the rules. But because I never thought that the plug-in I wrote could be officially included by coredns, so Just added this feature, I will remove it right away